ecobee/beam_batch.py

```diff
@@ -0,0 +1,144 @@
+import argparse
+import logging
+import os
+from collections import namedtuple
+from datetime import datetime
+from time import time
+
+import apache_beam as beam
+from apache_beam.io import ReadFromText
+from apache_beam.io import WriteToText
+from apache_beam.options.pipeline_options import PipelineOptions
+from apache_beam.options.pipeline_options import SetupOptions
+
+DataPoint = namedtuple("DataPoint", ["time", "action"])
+Sequence = namedtuple("Sequence", ["seq_len", "datapoints"])
+
+timestamp_prefix = "TS:"
+timestamp_postfix = "GMT "
+action_prefix = "Action:"
+date_fmt = "%Y-%m-%d %H:%M:%S %z"
+
+
+def to_data_point(item):
+    """
+    item =  TS:2018-12-03 13:29:15 +0000 GMT, Action:F
+    and converts it to DataPoint (2018-12-03 13:29:15 +0000, F) with first value as time (prefix and postfix removed)
+    and second value as just the action (prefix removed).
+    """
+    entry = item[: len(item) - 1].split(",") if item.endswith("\n") else item.split(",")
+    t, action = (
+        entry[0][len(timestamp_prefix) : -len(timestamp_postfix)],
+        entry[1][len(action_prefix) + 1 :],
+    )
+    return DataPoint(t, action)
```

Comment:
`t` is hard to read and doesnt tell anything about the underlying value its holding. we can re-name it `time_stamp`

Reply:
done